AWSutils.sh

```diff
@@ -167,9 +167,11 @@ function install_aws_utils()
 function InstallCLIv1 {
    local INSTALLDIR
    local BINDIR
+   local TMPDIR
 
    INSTALLDIR="/usr/local/aws-cli/v1"
    BINDIR="/usr/local/bin"
+   TMPDIR=$(chroot "${CHROOTMNT}" /bin/bash -c "mktemp -d")
```

Comment:
Better method (marginally more compact avoids pointless forking of subshells) would be:

~~~
bash -c "mktemp -p \"${CHROOTMNT}\" -d"
~~~

Example:
~~~
ferric@fountain:/tmp$ export CHROOTMNT=/var/tmp
ferric@fountain:/tmp$ bash -c "mktemp -p \"${CHROOTMNT}\" -d"
/var/tmp/tmp.IaNJiUAX39
~~~

Was originally going to recommend using `install`, but it looks like `mktemp` doesn't care what the invoking-shell's `umask` value is.

Note: the above also obviates the need for the `${CHROOTMNT}` token in your `${CHROOTMNT}${TMPDIR}` references later in your updates.

Reply:
Or, even shorter:

~~~
TMPDIR="$( mktemp -p "${CHROOTMNT}" -d )"
~~~

Reply:
I didn't want to do that because like you said, that includes `$CHROOTMNT` in the returned value. And later I need the TMPDIR value relative to the CHROOTMNT, when running the installer.